plan-for-extension-exercise/src/planForExtensionExercise/TotalAreaCalculator.java

```diff
@@ -0,0 +1,25 @@
+package planForExtensionExercise;
+
+public class TotalAreaCalculator extends AreaCalculator {
```

Comment:
Another argument in favor of making TotalAreaCalculator a subclass of AreaCalculator is that any code currently relying on AreaCalculator could be passed a TotalAreaCalculator instead, changing the behavior without changing the code that consumes it.

Of course, in an exercise like this there isn't enough context to make such decisions. But it's still fun to discuss.